Sources/Nats/NatsConnection.swift

```diff
@@ -497,16 +495,43 @@ class ConnectionHandler: ChannelInboundHandler {
     }
 
     func close() async throws {
+        self.reconnectTask?.cancel()
+        await self.reconnectTask?.value
         self.state = .closed
         try await disconnect()
         self.fire(.closed)
     }
 
-    func disconnect() async throws {
+    private func disconnect() async throws {
         self.pingTask?.cancel()
         try await self.channel?.close().get()
     }
 
+    func suspend() async throws {
+        self.reconnectTask?.cancel()
+        await self.reconnectTask?.value
+        if self.state == .connected {
+            self.state = .suspended
+            try await disconnect()
+            self.fire(.disconnected)
```

Comment:
Should suspended and disconnect be different events? even though suspension technically closes the connection from the clients perspective we are keeping subscription state whereas disconnect should also clear that state?

Reply:
We were discussing those.

We should never clear the state of subscriptions and we should always recover them when reconnecting/resuming.
The only difference is that if you disconnect because of suspend, client will not try to reconnect automatically. In all other cases, it will.

Reply:
I guess my point is about the event types more than the state. For example when you call `close()` we're emitting `closed` as opposed to `disconnected`

Reply:
Yeah, I was wondering weather to add a new event type or not. I was not sure how useful it would be to add a new event type, separate from "disconnected". Now that I think about it, having separate event types would mean you would be able to easily catch and separate instances of disconnects not triggered by the client and potentially take some action. Meanwhile, suspending is always initiated by the library and thus may not need a callback (or have a different one). I'll add the new event type.